lib/icinga/macroprocessor.cpp

```diff
@@ -111,17 +111,24 @@ bool MacroProcessor::ResolveMacro(const String& macro, const ResolverList& resol
 				if (!resolver.ResolveShortMacros)
 					continue;
 
+				Dictionary::Ptr vars;
 				CustomVarObject::Ptr dobj = dynamic_pointer_cast<CustomVarObject>(resolver.Obj);
 
 				if (dobj) {
-					Dictionary::Ptr vars = dobj->GetVars();
+					vars = dobj->GetVars();
+				} else {
+					IcingaApplication::Ptr dobj = dynamic_pointer_cast<IcingaApplication>(resolver.Obj);
```

Comment:
I don't even get what the `d` in original `dobj` is supposed to stand for, neither do I for the new variable. Given that it also shadows the other one, I'd chose a different name (maybe even for both, `cvObj` and `appObj` or something like that, good that there's already camel case and underscores mixed in that file).